spring-content-commons/src/main/java/org/springframework/content/commons/renditions/RenditionService.java

```diff
@@ -1,11 +1,14 @@
 package org.springframework.content.commons.renditions;
 
-import java.io.InputStream;
+import org.springframework.core.io.Resource;
 
 public interface RenditionService {
-	
-    public boolean canConvert(String fromMimeType, String toMimeType);
-    public String[] conversions(String fromMimeType);
-    public InputStream convert(String fromMimeType, InputStream fromInputSource, String toMimeType);
 
+	public boolean canConvert(String fromMimeType, String toMimeType);
+
+	public String[] conversions(String fromMimeType);
+
+	public Resource convert(String fromMimeType, Resource fromInputSource, String toMimeType);
+
+	public RenditionProvider getProvider(String fromMimeType, String toMimeType);
```

Comment:
I don't think this needs to be part of the "API".  It is only ever called by the implementation RenditionServiceImpl

Reply:
Actually I agree, Yes it can be removed from Interface.